src/server.h

```diff
@@ -3031,6 +3033,9 @@ void trimStringObjectIfNeeded(robj *o, int trim_small_values);
 static inline int canUseSharedObject(void) {
     return server.maxmemory == 0 || !(server.maxmemory_policy & MAXMEMORY_FLAG_NO_SHARED_INTEGERS);
 }
+static inline void updateMaxAvailableMemory(void) {
+    server.maxmemory_available = (unsigned long long)server.maxmemory / 100.0 * (100 - server.maxmemory_reserved_scale);
```

Comment:
If we are aligned on the config name `maxmemory_soft_scale`, I would suggest going with `maxmemory_soft` for the field name too
```suggestion
    server.maxmemory_soft = (unsigned long long)server.maxmemory / 100.0 * (100 - server.maxmemory_soft_scale);
```